iroha/src/block_sync.rs

```diff
@@ -63,7 +115,7 @@ impl BlockSynchronizer {
         let gossip_period = self.gossip_period;
         let wsv = Arc::clone(&self.wsv);
         let peer_id = self.peer_id.clone();
-        let sumeragi = Arc::clone(&self.sumeragi);
+        let sumeragi = self.sumeragi.clone();
         drop(task::spawn(
             async move {
                 loop {
```

Comment:
I thought you wanted to not spawn tasks but implement this kinds of loops by sending message to self?